Public/Src/Engine/Processes/SharedOpaqueOutputHelper.cs

```diff
@@ -134,34 +174,13 @@ public static bool IsSharedOpaqueOutput(string expandedPath)
         /// <exception cref="BuildXLException">When unsuccessful</exception>
         public static void SetPathAsSharedOpaqueOutput(string expandedPath)
         {
-            // In the case of a no replay, this case can happen if the file got into the cache as a static output,

```

Comment:
There was a reason why before any of my changes this was only being called when running on Windows: the implementation for Unix simply does something else; there are some cache components that depend on that behavior so I couldn't just changed the implementation for Unix, so instead I implemented it in `Unix.SetPathAsSharedOpaqueOutput`